pcdsdevices/utils.py

```diff
@@ -687,3 +690,151 @@ def post_ophyds_to_elog(objs, allow_child=False, hutch_elog=None):
 
     hutch_elog.post(final_post, tags=['ophyd_status'],
                     title='ophyd status report')
+
+
+def reorder_components(
+    cls: type,
+    start_with: Optional[List[Union[str, Cpt]]] = None,
+    end_with: Optional[List[Union[str, Cpt]]] = None,
+) -> None:
+    """
+    Rearrange the components in cls for typhos displays.
+
+    Internally, this works by switching around the keys in the _sig_attrs
+    OrderedDict.
+
+    Parameters
+    ----------
+    cls : type
+        The Device subclass that we'd like to rearrange the order of.
+    start_with : list of str, optional
+        The component names to bring to the top of the screen.
+    end_with : list of str, optional
+        The component names to bring to the bottom of the screen.
+    """
+    start_with = _normalize_reorder_list(cls, start_with)
+    end_with = _normalize_reorder_list(cls, end_with)
+    for cpt_name in reversed(start_with):
+        cls._sig_attrs.move_to_end(cpt_name, last=False)
+    for cpt_name in end_with:
+        cls._sig_attrs.move_to_end(cpt_name, last=True)
+
+
+def _normalize_reorder_list(
+    cls: type,
+    cpts_or_names: Optional[List[Union[str, Cpt]]],
+) -> List[str]:
+    """
+    Simplify the user's variable arguments for the component reordering.
+    """
+    if cpts_or_names is None:
+        return []
+    reverse_map = {cpt: name for name, cpt in cls._sig_attrs.items()}
+    output = []
+    for obj in cpts_or_names:
+        if isinstance(obj, Cpt):
+            output.append(reverse_map[obj])
```

Comment:
Maybe: `except KeyError`, and raise an explanatory "you provided a component not in the same class, what are you doing?"

Reply:
Good call